### PR TITLE
fix promoter manifests to the new 1.0+ syntax

### PR DESCRIPTION
**Pull request by listx**

### CODE REVIEW

**Comment by listx** [comment]
OK so the pull-k8sio-cip job is running (woohoo!!!), but the business logic around the git delta is failing to see that the manifest has changed. Gonna start debugging this after I eat lunch... I'll be back later.

**Comment by listx** [comment]
Hmm also we should make `pull-k8sio-cip` a *Required* test (should be a one-liner change somewhere in our infra, is my guess).

**Comment by thockin** [comment]
We should probably have a staging and prod repo JUST for testing image promoter.

**Comment by listx** [comment]
OK I found a number of small issues in the promoter. Gonna update that, update the prow job, and then refresh this PR.

**Comment by listx** [comment]
Awesome! It correctly failed complaining that the image is not found https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/k8s.io/212/pull-k8sio-cip/1113250066883153921

Now I need to modify this PR to have the correct digest.

**Comment by listx** [comment]
TODO:

1. Make Prow mark the `pull-k8sio-cip` job as `Required`. This may require the branch protection bot to run.
2. Test some bad changes to this PR
- introduce a typo in the digest to invoke the promoter to exit with an invalid digest error
- introduce a typo in the tag
- introduce a typo in the YAML (yaml parsing error should occur)
3. Update the {ci,post}-k8sio-cip jobs in Prow accordingly to use the updated (fixed) configuration in the `pull-k8sio-cip` job that we are testing in this PR.

**Comment by listx** [comment]
Update:

Very close. 17517fe should pass, but it is correctly complaining in https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/k8s.io/212/pull-k8sio-cip/1113666785804029952 that gcr.io/k8s-staging-csi/coredns@sha256:02382353821b12c21b062c59184e227e001079bb13ebd01f9d3270ba0fcbf1e4 is missing. Since @dims added that image to the original manifest, I think it should be there and we should use that to test the promotion. If we agree, then I can make 1 more change to the Prow configs so that the postsumbit promoter run (actual promoter run) will promote coredns.

One other thing I forgot but must confirm: does `gcr.io/k8s-staging` exist? The promoter is complaining that it can't read from it (see the prow status link above), although I'll paste a portion here for convenience:

```
Request {gcr.io/k8s-staging 0xc00009c4c0}: error(s) encountered: [{process had stderr ERROR: (gcloud.container.images.list) Bad status during token exchange: 400
{"errors":[{"code":"UNKNOWN","message":"Project 'projects/k8s-staging' not found or deleted."}]}
} {closing process exit status 1}]
```

**Comment by thockin** [comment]
Sounds great!  Thanks for the update.

There is no k8s-staging.  There is k8s-staging-cordens and
k8s-staging-cluster-api and so forth.  Where is it getting k8s-staging with
no suffix?

**Comment by listx** [comment]
The k8s-staging without suffix is here: https://github.com/kubernetes/k8s.io/blob/master/k8s.gcr.io/k8s-staging-coredns/manifest.yaml#L4

Err wait I guess it should just be gcr.io/k8s-gcr-prod right?

**Comment by thockin** [comment]
```
$ gcloud projects list   --filter "parent.id=758905017065"   --format
"value(name)"
k8s-cip-test-prod
k8s-staging-cip-test
k8s-gcr-graveyard
k8s-gcr-prod
k8s-staging-csi
k8s-staging-cluster-api
k8s-staging-coredns
kubernetes-public
```

**Comment by listx** [comment]
Ok ya, so the manifests are wrong. I'm going to:
(1) update Prow so that the trusted (real promotion!!) jobs will take effect
(2) update this PR to fix the manifests to promote to `gcr.io/k8s-gcr-prod` (and not the dubious `gcr.io/k8s-staging`)
(3) make sure the gcr.io/k8s-staging-csi/coredns@sha256:02382353821b12c21b062c59184e227e001079bb13ebd01f9d3270ba0fcbf1e4 image exists
(4) ask reviewers to LGTM this PR --- at this point, because of (1), we should see the coredns image in (3) get promoted to `gcr.io/k8s-gcr-prod`.

After this, we can start thinking about adding the e2e test (I'll be on this), backups and other issues (garbage collection, etc). These and other issues have already been documented in the issue tracker for the promoter [here](https://github.com/kubernetes-sigs/k8s-container-image-promoter/issues), so hopefully we can work in parallel...

**Comment by dims** [comment]
@listx - i used `docker pull k8s.gcr.io/coredns:1.3.1` to fetch the image

```
[dims@dims-mac 19:27] ~/go/src/k8s.io/kubernetes ⟩ docker pull k8s.gcr.io/coredns:1.3.1
1.3.1: Pulling from coredns
e0daa8927b68: Pull complete
3928e47de029: Pull complete
Digest: sha256:02382353821b12c21b062c59184e227e001079bb13ebd01f9d3270ba0fcbf1e4
Status: Downloaded newer image for k8s.gcr.io/coredns:1.3.1
```

**Comment by listx** [comment]
OK so (1), (2), and (3) are done. So we are at step (4).

Also, the multirun.sh is correctly invoking the promoter for all 3 manifests because this PR touched all 3, as you can see https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/k8s.io/212/pull-k8sio-cip/1113962517811433472/build-log.txt (hard to parse, but it did run 3 times).

I'm going to make a couple changes to the promoter to make it easier to parse the 3 runs from the Prow outputs, and also update this PR to test multirun.sh again (e.g., I want to make it only run on 1 of the manifests, not all 3).

Stay tuned.

**Comment by listx** [comment]
So the multirun.sh correctly only ran against 2 manifests! I'm going to squash these commits and fix up this PR for final approval.

Meanwhile let's wait for https://github.com/kubernetes/test-infra/pull/12085 to get merged first, because that one will give us more clearly delineated log output in the post run (after this PR is merged).

**Comment by listx** [comment]
Yay so https://github.com/kubernetes/test-infra/pull/12085 was merged.

/retest

**Comment by listx** [comment]
OK things are looking really good. Please click on `Details` next to `pull-k8sio-cip` to see the latest run. You can see how the promoter ran 3 times against all 3 changed manifests in this PR.

@dims can you approve?

/hold cancel

**Comment by listx** [comment]
The post-k8sio-cip job correctly fired after this was merged, but the run crashed because gcloud crashed. Logs are https://storage.googleapis.com/kubernetes-jenkins/logs/post-k8sio-cip/1114110365160968199/build-log.txt

The line that crashed had to do with [gcloud credential revocation](https://github.com/kubernetes-sigs/k8s-container-image-promoter/blob/master/util/multirun.sh#L152). Interestingly, it looks like there is at least 1 other user who is experiencing this same issue since last month: https://stackoverflow.com/questions/55127645/gcloud-crashed-tokenrevokeerror-invalid-request.

I've tried to reproduce locally, but my gcloud version 241.0.0 says this:
```
WARNING: [k8s-infra-gcr-promoter@k8s-gcr-prod.iam.gserviceaccount.com] appears to be a service account. Service account tokens cannot be revoked, but they will expire automatically. To prevent use of the service account token earlier than the expiration, revoke the parent service account or service account key.
```
and does not crash. I'm going to update the gcloud version in the bazel rules to fix this.

**Comment by listx** [comment]
I opened https://github.com/kubernetes/test-infra/pull/12090 to get around the issue. After that gets merged I need to re-run the failed [post-k8sio-cip job](https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/post-k8sio-cip/1114110365160968199).
